### PR TITLE
Day of the week

### PR DESCRIPTION
**Pull request by denisvieira-dev**

### CODE REVIEW

**Review by gjorgji-grgovski-authoritypartners — CHANGES_REQUESTED** [review]

**Comment by gjorgji-grgovski-authoritypartners** [comment]
Please rollback all deleted files

**Review by gjorgji-grgovski-authoritypartners — CHANGES_REQUESTED** [review]

File: src/Nox.Types/Enums/DayOfTheWeek.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+﻿namespace Nox.Enums;
+
+public enum DayOfTheWeek
```
[the remark this answers is not in the document]
Comment (denisvieira-dev) [reply src/Nox.Types/Enums/DayOfTheWeek.cs]: Done

File: src/Nox.Types/Types/DayOfWeek/DayOfWeek.cs [hunk 2]
```diff
@@ -1,9 +1,85 @@
...
 {
+    public DayOfWeek() { Value = 0; }
+
+    /// <summary>
+    /// Creates a new instance of <see cref="DayOfWeek"/>
+    /// </summary>
+    /// <param name="value">The number to create the <see cref="DayOfWeek"/> with</param>
+    /// <returns></returns>
+    /// <exception cref="ValidationException"></exception>
+    public static new DayOfWeek From(int value)
```
[the remark this answers is not in the document]
Comment (denisvieira-dev) [reply src/Nox.Types/Types/DayOfWeek/DayOfWeek.cs]: Done

File: src/Nox.Types/Types/DayOfWeek/DayOfWeek.cs [hunk 3]
```diff
@@ -1,9 +1,85 @@
-﻿namespace Nox.Types;
+﻿using Nox.Enums;
+
+namespace Nox.Types;
 
 /// <summary>
 /// Represents a Nox <see cref="DayOfWeek"/> type and value object.
 /// </summary>
 /// <remarks>Placeholder, needs to be implemented</remarks>
 public sealed class DayOfWeek : ValueObject<int, DayOfWeek>
```
[the remark this answers is not in the document]
Comment (denisvieira-dev) [reply src/Nox.Types/Types/DayOfWeek/DayOfWeek.cs:8]: Done

File: src/Nox.Types/Types/DayOfWeek/DayOfWeek.cs [hunk 4]
```diff
@@ -1,9 +1,61 @@
...
 {
+    public DayOfWeek() { Value = 0; }
+
+    public static DayOfWeek From(DayOfWeek dayOfTheWeek)
```
[the remark this answers is not in the document]
Comment (denisvieira-dev) [reply src/Nox.Types/Types/DayOfWeek/DayOfWeek.cs:12]: Done

**Review by gjorgji-grgovski-authoritypartners — CHANGES_REQUESTED** [review]

**Review by gjorgji-grgovski-authoritypartners — APPROVED** [review]

File: src/Nox.Types.Abstractions/Enums/DayOfTheWeek/DayOfTheWeek.cs
Comment (rochar) [inline src/Nox.Types.Abstractions/Enums/DayOfTheWeek/DayOfTheWeek.cs:1]: I would use System.DayOfWeek and avoid add anew type

File: src/Nox.Types/Types/DayOfWeek/DayOfWeek.cs [hunk 5]
```diff
@@ -1,9 +1,74 @@
...
+        }
+
+        if (Value > 6)
+        {
+            result.Errors.Add(new ValidationFailure(nameof(Value), $"Could not create a Nox DayOfWeek type a value {Value} is greater than the maximum specified value of 6"));
+        }
+
+        return result;
+    }
+
```
Comment (rochar) [inline src/Nox.Types/Types/DayOfWeek/DayOfWeek.cs:40]: Add to WeekDay() and return System.DayOfWeek from the value

**Review by rochar — CHANGES_REQUESTED** [review]
Looks good, just a small comment

**Review by rochar — APPROVED** [review]

**Review by rochar — APPROVED** [review]

**Pull request merged**